### PR TITLE
Integration tests for requirejs implementation

### PR DESCRIPTION
**Pull request by magska**

### CODE REVIEW

File: dev/main.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+'use strict';
+/* global requirejs */
+/* global define */
+
+requirejs.config({
+	paths: {
+		jquery: '//ajax.googleapis.com/ajax/libs/jquery/2.1.3/jquery.min',
```
Comment (rob-mccann) [inline dev/main.js]: I'd advise against requiing use of jquery for something that's going to get used everywhere and needs to be performant; is it strictly necessary?

File: dev/require.js [hunk 2]
```diff
@@ -0,0 +1,2083 @@
...
+ * @license RequireJS 2.1.17 Copyright (c) 2010-2015, The Dojo Foundation All Rights Reserved.
+ * Available via the MIT or new BSD license.
+ * see: http://github.com/jrburke/requirejs for details
+ */
+//Not using strict: uneven strict support in browsers, #392, and causes
+//problems with requirejs.exec()/transpiler plugins that may not be strict.
+/*jslint regexp: true, nomen: true, sloppy: true */
+/*global window, navigator, document, importScripts, setTimeout, opera */
+
+var requirejs, require, define;
```
Comment (rob-mccann) [inline dev/require.js]: should this be install by a package manager? Is it minified in your pipeline?

File: dev/main.js (hunk 1 above)
Comment (magska) [inline dev/main.js]: This is just a test, added jquery just to get some more scripts in there.

File: test/integration/testPageload.js [hunk 3]
```diff
@@ -5,49 +5,64 @@ module.exports = {
...
-            this.assert.equal(result.value[2].name, '_DataTrackerVisitor');
-            chai.assert.match(result.value[2].value, uuidv4pattern);
-            this.assert.equal(result.value[3].name, '_DataTrackerUser');
-            chai.assert.equal(result.value[3].value, 'undefined');
+			this.assert.equal(result.value.length, 4);
+
+			var count = 0;
+
+			for (var j = 0; j < result.value.length; j++) {
+				if (result.value[j].name === '_DataTrackerSession') {
```
Comment (rob-mccann) [inline test/integration/testPageload.js]: having if statements in tests makes me a little nervous - what are you testing here?

Comment (magska) [inline test/integration/testPageload.js]: I am testing that a specific cookie key is present, and that each key are present 1 time, and that they have a value that is somewhat correct.

File: test/integration/testRequireJS.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
+/* global server */
+/* global AutoTrack */
+
+'use strict';
+
+module.exports = {
+    'Pageload with requirejs': function (browser) {
+        var chai = require('chai');
```
Comment (rob-mccann) [inline test/integration/testRequireJS.js]: require at the top of the page

File: dev/main.js (hunk 1 above)
Comment (rob-mccann) [inline dev/main.js]: after chatting - looks like this is just here to check it's working alongside jquery rather than depending on it

**Pull request merged**